### PR TITLE
Modernize python versions

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: translationstring/tests/test__init__.py [hunk 1]
```diff
@@ -77,56 +75,56 @@ def test_interpolate_substitution(self):
         inst = self._makeOne('This is $name version ${version}.',
                               mapping=mapping)
         result = inst.interpolate()
-        self.assertEqual(result, u('This is Zope version 3.'))
+        self.assertEqual(result, u'This is Zope version 3.')
```
Comment (davisagli) [inline translationstring/tests/test__init__.py:78]:
```suggestion
        self.assertEqual(result, 'This is Zope version 3.')
```

& similar below (pyupgrade could help clean this up)

File: docs/glossary.rst [hunk 2]
```diff
@@ -21,13 +21,19 @@ Glossary
      predefines a :term:`translation domain`.
 
    Translation Domain
-     A string representing the "context" in which a particular
+     A string representing the "domain" in which a particular
      translation was made.  For example the word "java" might be
      translated differently if the translation domain is
      "programming-languages" than would be if the translation domain
      was "coffee".  Every :term:`translation string` has an associated
```
[the remark this answers is not in the document]
Comment (tseaver) [reply docs/glossary.rst]: @davisagli Thanks for the catch!  I agree:  751dc78

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ jobs:
 
     strategy:
       matrix:
-        python: ["2.7", "3.6", "3.7", "3.8"]
+        python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
[the remark this answers is not in the document]
Comment (tseaver) [reply .github/workflows/main.yaml]: 688a395

File: translationstring/__init__.py [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
-from translationstring.compat import string_types
-from translationstring.compat import PY3
 
 NAME_RE = r"[a-zA-Z][-a-zA-Z0-9_]*"
 
 _interp_regex = re.compile(r'(?<!\$)(\$(?:(%(n)s)|{(%(n)s)}))'
     % ({'n': NAME_RE}))
 
-CONTEXT_MASK = text_type('%s\x04%s')
+CONTEXT_MASK = u'%s\x04%s'
```
[the remark this answers is not in the document]
Comment (tseaver) [reply translationstring/__init__.py]: There were a bunch in the tests, too.  75d6c95

File: translationstring/tests/test__init__.py (hunk 1 above)
Comment (tseaver) [reply translationstring/tests/test__init__.py]: 75d6c95

**Review by davisagli — APPROVED** [review]

**Pull request merged**